bindings/node/test/autoEncrypter.test.js

```diff
@@ -143,6 +143,27 @@ describe('AutoEncrypter', function () {
       });
     });
 
+    it('should decrypt mock data and mark decrypted items if enabled for testing', function (done) {
+      const input = readExtendedJsonToBuffer(`${__dirname}/data/encrypted-document.json`);
+      const client = new MockClient();
+      const mc = new AutoEncrypter(client, {
+        keyVaultNamespace: 'admin.datakeys',
+        logger: () => {},
+        kmsProviders: {
+          aws: { accessKeyId: 'example', secretAccessKey: 'example' },
+          local: { key: Buffer.alloc(96) }
+        }
+      });
+      mc[Symbol.for('@@mdb.decorateDecryptionResult')] = true;
+      mc.decrypt(input, (err, decrypted) => {
+        if (err) return done(err);
+        expect(decrypted).to.eql({ filter: { find: 'test', ssn: '457-55-5462' } });
+        expect(decrypted[Symbol.for('@@mdb.decryptedKeys')]).to.eql(undefined);
```

Comment:
The current implementation will define the array only if it is needed, so maybe this assertion could be explicit about that:
```js
expect(decrypted).to.not.have.property(Symbol.for('@@mdb.decryptedKeys'));
```

But it raises the question would it be more convenient to define it always so `.length` is always safe to check? but then again maybe that's unnecessary overhead, thoughts?

Reply:
Sure, works for me.

> But it raises the question would it be more convenient to define it always so `.length` is always safe to check? but then again maybe that's unnecessary overhead, thoughts?

I’m not so much worried about performance here, but I’m a little hesitant to make the presence of the property for all objects something that consuming code should rely on, tbh.

Reply:
Yea that makes sense to me too. Semi-related, should the array be frozen after being filled? I think it would be important that no keys get removed, although it is a feature used in the more or less locked ecosystem of shell/compass, seems safer to guarantee finality at this layer. overkill?

Reply:
It seems a bit like overkill, yeah. I don’t see why we would *want* to modify it, but we might at some point do that in Compass, and conversely, any code which would be prevented from updating the array would almost certainly be able to get around that by just updating the entire object tree .

Reply:
Very true, sounds like we've got what we need here. Ty!